src/olympia/abuse/actions.py

```diff
@@ -383,10 +383,23 @@ class ContentActionRejectVersionDelayed(ContentActionRejectVersion):
 
     def __init__(self, decision):
         super().__init__(decision)
-        self.days = int(self.decision.metadata.get('delayed_rejection_days', 0))
+        if 'delayed_rejection_date' in self.decision.metadata:
+            self.delayed_rejection_date = datetime.fromisoformat(
+                self.decision.metadata.get('delayed_rejection_date')
+            )
+        else:
```

Comment:
Is the implicit assumption here that if the value actually is none, then later methods will raise? We should just raise explicitly here then no?

Reply:
My comment is probably poorly worded. There are times when we need to instantiate the class and call some methods that don't need the `delayed_rejection_date`, and we want those to continue to work.

An example is found in `TestCinderJob.test_report_with_outstanding_rejection` which sends an email to the reporter of an abuse report filed against an add-on with a delayed rejection already set (but that delayed rejection decision doesn't have that metadata - as might be the case for existing ones before this change)

Reply:
Just curious if that could have been resolved with a migration? All good though no need for a change.

Reply:
Maybe. We haven't tried to do that in the past, it's a bit of an edge case (we typically don't have to worry about executing past decisions...)